eva-accession-clustering/src/main/java/uk/ac/ebi/eva/accession/clustering/batch/io/StudyClusteringMongoReader.java

```diff
@@ -0,0 +1,115 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.ac.ebi.eva.accession.clustering.batch.io;
+
+import com.mongodb.BasicDBObject;
+import com.mongodb.client.FindIterable;
+import com.mongodb.client.MongoCursor;
+import com.mongodb.client.model.Filters;
+import org.bson.Document;
+import org.bson.conversions.Bson;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.batch.item.ExecutionContext;
+import org.springframework.batch.item.ItemStreamException;
+import org.springframework.batch.item.ItemStreamReader;
+import org.springframework.data.mongodb.core.MongoTemplate;
+import org.springframework.data.mongodb.core.convert.MongoConverter;
+
+import uk.ac.ebi.eva.accession.core.model.eva.SubmittedVariantEntity;
+
+import java.util.List;
+
+public class StudyClusteringMongoReader implements ItemStreamReader<SubmittedVariantEntity> {
+
+    private static final Logger logger = LoggerFactory.getLogger(StudyClusteringMongoReader.class);
+
+    static final String ASSEMBLY_FIELD = "seq";
+
+    static final String CLUSTERED_VARIANT_ACCESSION_FIELD = "rs";
+
+    static final String STUDY_FIELD = "study";
+
+    String assembly;
+
+    List<String> studies;
+
+    MongoCursor<Document> evaCursor;
+
+    MongoConverter converter;
+
+    MongoTemplate mongoTemplate;
+
+    int chunkSize;
+
+    public StudyClusteringMongoReader(MongoTemplate mongoTemplate, String assembly, List<String> studies,
+                                      int chunkSize) {
+        this.mongoTemplate = mongoTemplate;
+        this.assembly = assembly;
+        this.studies = studies;
+        this.chunkSize = chunkSize;
+    }
+
+    @Override
+    public SubmittedVariantEntity read() {
+        Document nextElement = evaCursor.tryNext();
+        return (nextElement != null) ? getSubmittedVariantEntity(nextElement) : null;
+    }
+
+    private SubmittedVariantEntity getSubmittedVariantEntity(Document notClusteredSubmittedVariants) {
+        return converter.read(SubmittedVariantEntity.class, new BasicDBObject(notClusteredSubmittedVariants));
+    }
+
+    @Override
+    public void open(ExecutionContext executionContext) throws ItemStreamException {
+        initializeReader();
+    }
+
+    public void initializeReader() {
+        evaCursor = initializeCursor();
+        converter = mongoTemplate.getConverter();
+    }
+
+    private MongoCursor<Document> initializeCursor() {
+        Bson query = Filters.and(Filters.in(ASSEMBLY_FIELD, assembly),
+                                 Filters.in(STUDY_FIELD, studies),
+                                 Filters.exists(CLUSTERED_VARIANT_ACCESSION_FIELD, false),
```

Comment:
I checked and [$exists does used the index](https://jira.mongodb.org/browse/SERVER-393), so it should be fast enough unless there is a massive study that we're not clustering.